label_studio/core/utils/db.py

```diff
@@ -23,3 +23,10 @@ def fast_first(queryset: QuerySet[ModelType]) -> Optional[ModelType]:
     if result := queryset[:1]:
         return result[0]
     return None
+
+
+def fast_first_or_create(model, **model_params) -> Optional[ModelType]:
+    """Like get_or_create, but using fast_first instead of first()"""
```

Comment:
I would add 
```suggestion
    """Like get_or_create, but using fast_first instead of first(). Additionally, unlike get_or_create, this method will not raise an exception if more than one model instance matching the given params is returned, making it a safer choice than get_or_create for models that don't have a uniqueness constraint on the fields used."""
```